File: backend/src/main/java/todoktodok/backend/book/application/dto/response/BookResponse.java
package todoktodok.backend.book.application.dto.response;

import todoktodok.backend.book.domain.Book;

public record BookResponse(
        Long bookId,
        String bookTitle,
        String bookAuthor,
        String bookImage,
        String bookPublisher,
        String bookSummary
) {

    public BookResponse(final Book book) {
        this(
                book.getId(),
                book.getTitle(),
                book.getAuthor(),
                book.getImage(),
                book.getPublisher(),
                book.getSummary()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/BookResponse.kt
package com.team.todoktodok.data.network.response

import com.team.domain.model.Book
import kotlinx.serialization.Serializable

@Serializable
data class BookResponse(
    val bookId: Long,
    val bookTitle: String,
    val bookAuthor: String,
    val bookImage: String,
) {
    fun toDomain(): Book =
        Book(
            id = this.bookId,
            title = this.bookTitle,
            author = this.bookAuthor,
            image = this.bookImage,
        )
}
